src/util/ui/widgets.rs

```diff
@@ -30,56 +50,72 @@ impl<'w, 's> MyWidgets<'w> for Commands<'w, 's> {
             ..default()
         })
     }
+}
+
+/// Widgets spawning trait.
+pub(crate) trait Widgets<'a> {
+    /// Spawns a simple button node with text.
+    fn button<I: Into<String>>(&mut self, text: I) -> EntityCommands;
 
-    fn my_button<I: Into<String>>(&mut self, text: I) -> EntityCommands {
-        let button = self
-            .spawn((ButtonBundle {
+    /// Spawns a simple label.
+    fn label<I: Into<String>>(&mut self, text: I) -> EntityCommands;
+}
+
+impl<'a, T: Spawner<'a>> Widgets<'a> for T {
+    fn button<I: Into<String>>(&mut self, text: I) -> EntityCommands {
+        let mut entity_commands = self.spawn((
```

Comment:
This could be a single method chain with `self.spawn(...).with_children(...)`. Same for `fn label`.

Reply:
no it cannot, with_children returns mutable reference not ownership, i tried :(

Reply:
Unfortunate.